抓取Discuz! X3.4板块出现错误

### ELF_RSS 、 go-cqhttp 、 nonebot 、 Python 版本及操作系统

nonebot 2.0.0b5
python 3.10.1
win10 ltsc
ELF-RSS v2.6.12

### 列出安装的 Python 包

_No response_

### 如何复现

@bot add 悠闲数学娱乐论坛 http://kuing.infinityfreeapp.com/forum.php?mod=rss&fid=5&auth=0

### 期望行为

_No response_

### 实际行为

10-24 03:01:28 [INFO] ELF_RSS | 悠闲数学娱乐论坛 检查更新
10-24 03:01:29 [ERROR] apscheduler | Job "check_update (trigger: interval[0:05:00], next run at: 2022-10-24 03:06:36 CST)" raised an exception
Traceback (most recent call last):
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\Scripts\nb.exe\__main__.py", line 7, in <module>
    sys.exit(main())
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\click\core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\nb_cli\commands\main.py", line 30, in run
    run_bot(file, app)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\nb_cli\handlers\deploy.py", line 25, in run_bot
    nonebot.run(app=f"{module_name}:{app}")
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\nonebot\__init__.py", line 261, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\nonebot\drivers\fastapi.py", line 170, in run
    uvicorn.run(
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\uvicorn\main.py", line 576, in run
    server.run()
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\uvicorn\server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 633, in run_until_complete
    self.run_forever()
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 600, in run_forever
    self._run_once()
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1896, in _run_once
    handle._run()
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
  File "C:\Users\Alex\Documents\LittlePaimon\.\ELF_RSS\src\plugins\ELF_RSS2\my_trigger.py", line 20, in check_update
    await asyncio.wait_for(rss_parsing.start(rss), timeout=wait_for)
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\tasks.py", line 445, in wait_for
    return fut.result()
  File "C:\Users\Alex\Documents\LittlePaimon\.\ELF_RSS\src\plugins\ELF_RSS2\rss_parsing.py", line 95, in start
    await pr.start(rss_name=rss.name, new_rss=new_rss)
  File "C:\Users\Alex\Documents\LittlePaimon\.\ELF_RSS\src\plugins\ELF_RSS2\parsing\parsing_rss.py", line 158, in start
    rss_title = new_rss["feed"]["title"]
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\feedparser\util.py", line 113, in __getitem__
    return dict.__getitem__(self, key)
KeyError: 'title'
2022-10-24 03:01:29 ERROR Job "check_update (trigger: interval[0:05:00], next run at: 2022-10-24 03:06:36 CST)" raised an exception
Traceback (most recent call last):
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
  File "C:\Users\Alex\Documents\LittlePaimon\.\ELF_RSS\src\plugins\ELF_RSS2\my_trigger.py", line 20, in check_update
    await asyncio.wait_for(rss_parsing.start(rss), timeout=wait_for)
  File "C:\Users\Alex\AppData\Local\Programs\Python\Python310\lib\asyncio\tasks.py", line 445, in wait_for
    return fut.result()
  File "C:\Users\Alex\Documents\LittlePaimon\.\ELF_RSS\src\plugins\ELF_RSS2\rss_parsing.py", line 95, in start
    await pr.start(rss_name=rss.name, new_rss=new_rss)
  File "C:\Users\Alex\Documents\LittlePaimon\.\ELF_RSS\src\plugins\ELF_RSS2\parsing\parsing_rss.py", line 158, in start
    rss_title = new_rss["feed"]["title"]
  File "C:\Users\Alex\AppData\Local\pypoetry\Cache\virtualenvs\littlepaimon-Sqv748AN-py3.10\lib\site-packages\feedparser\util.py", line 113, in __getitem__
    return dict.__getitem__(self, key)
KeyError: 'title'

--- a/src/plugins/ELF_RSS2/__init__.py
+++ b/src/plugins/ELF_RSS2/__init__.py
@@ -11,6 +11,9 @@ from .config import DATA_PATH
 from .rss_class import Rss
 from .utils import send_message_to_admin
 
+require("nonebot_plugin_apscheduler")
+from nonebot_plugin_apscheduler import scheduler  # noqa
+
 VERSION = "2.6.12"
 __plugin_meta__ = PluginMetadata(
     name="ELF_RSS",
@@ -18,8 +21,6 @@ __plugin_meta__ = PluginMetadata(
     usage="https://github.com/Quan666/ELF_RSS",
     extra={"author": "Quan666 <i@Rori.eMail>", "version": VERSION},
 )
-
-scheduler = require("nonebot_plugin_apscheduler").scheduler
 
 
 async def check_first_connect(_: LifecycleMetaEvent) -> bool:

--- a/src/plugins/ELF_RSS2/command/add_cookies.py
+++ b/src/plugins/ELF_RSS2/command/add_cookies.py
@@ -48,10 +48,6 @@ async def handle_add_cookies(rss_cookies: str = ArgPlainText("COOKIES")) -> None
         await ADD_COOKIES.finish(f"❌ 不存在该订阅: {name}")
     else:
         rss.name = name
-        if rss.set_cookies(cookies):
-            await tr.add_job(rss)
-            await ADD_COOKIES.finish(f"👏 {rss.name}的Cookies添加成功！")
-        else:
-            await ADD_COOKIES.finish(
-                f"❌ {rss.name}的Cookies添加失败！\nCookies:{rss.cookies}"
-            )
+        rss.set_cookies(cookies)
+        await tr.add_job(rss)
+        await ADD_COOKIES.finish(f"👏 {rss.name}的Cookies添加成功！")

--- a/src/plugins/ELF_RSS2/my_trigger.py
+++ b/src/plugins/ELF_RSS2/my_trigger.py
@@ -44,9 +44,7 @@ def rss_trigger(rss: Rss) -> None:
         my_trigger_cron(rss)
         return
     # 制作一个“time分钟/次”触发器
-    trigger = IntervalTrigger(
-        minutes=int(rss.time), jitter=10, timezone="Asia/Shanghai"
-    )
+    trigger = IntervalTrigger(minutes=int(rss.time), jitter=10)
     # 添加任务
     scheduler.add_job(
         func=check_update,  # 要添加任务的函数，不要带参数
@@ -81,7 +79,6 @@ def my_trigger_cron(rss: Rss) -> None:
             day=times_list[2],
             month=times_list[3],
             day_of_week=times_list[4],
-            timezone="Asia/Shanghai",
         )
     except Exception:
         logger.exception(f"创建定时器错误！cron:{times_list}")

--- a/src/plugins/ELF_RSS2/my_trigger.py
+++ b/src/plugins/ELF_RSS2/my_trigger.py
@@ -10,6 +10,9 @@ from nonebot.log import logger
 from . import rss_parsing
 from .rss_class import Rss
 
+require("nonebot_plugin_apscheduler")
+from nonebot_plugin_apscheduler import scheduler  # noqa
+
 wait_for = 5 * 60
 
 
@@ -23,7 +26,6 @@ async def check_update(rss: Rss) -> None:
 
 
 def delete_job(rss: Rss) -> None:
-    scheduler = require("nonebot_plugin_apscheduler").scheduler
     if scheduler.get_job(rss.name):
         scheduler.remove_job(rss.name)
 
@@ -41,7 +43,6 @@ def rss_trigger(rss: Rss) -> None:
     if re.search(r"[_*/,-]", rss.time):
         my_trigger_cron(rss)
         return
-    scheduler = require("nonebot_plugin_apscheduler").scheduler
     # 制作一个“time分钟/次”触发器
     trigger = IntervalTrigger(
         minutes=int(rss.time), jitter=10, timezone="Asia/Shanghai"
@@ -85,7 +86,6 @@ def my_trigger_cron(rss: Rss) -> None:
     except Exception:
         logger.exception(f"创建定时器错误！cron:{times_list}")
         return
-    scheduler = require("nonebot_plugin_apscheduler").scheduler
 
     # 添加任务
     scheduler.add_job(

--- a/src/plugins/ELF_RSS2/parsing/__init__.py
+++ b/src/plugins/ELF_RSS2/parsing/__init__.py
@@ -1,8 +1,6 @@
 import re
 import sqlite3
-from contextlib import suppress
 from difflib import SequenceMatcher
-from email.utils import parsedate_to_datetime
 from typing import Any, Dict, List
 
 import arrow
@@ -19,7 +17,7 @@ from .cache_manage import (
     insert_into_cache_db,
     write_item,
 )
-from .check_update import check_update
+from .check_update import check_update, get_item_date
 from .download_torrent import down_torrent, pikpak_offline
 from .handle_html_tag import handle_html_tag
 from .handle_images import handle_img
@@ -325,12 +323,8 @@ async def handle_date(
     tmp: str,
     tmp_state: Dict[str, Any],
 ) -> str:
-    if date := item.get("published", item.get("updated")):
-        with suppress(Exception):
-            date = parsedate_to_datetime(date)
-        date = arrow.get(date).to("Asia/Shanghai")
-    else:
-        date = arrow.now()
+    date = get_item_date(item)
+    date = date.replace(tzinfo="local") if date > arrow.now() else date.to("local")
     return f"日期：{date.format('YYYY年MM月DD日 HH:mm:ss')}"
 
 

--- a/src/plugins/ELF_RSS2/qbittorrent_download.py
+++ b/src/plugins/ELF_RSS2/qbittorrent_download.py
@@ -20,6 +20,9 @@ from .utils import (
     send_message_to_admin,
 )
 
+require("nonebot_plugin_apscheduler")
+from nonebot_plugin_apscheduler import scheduler  # noqa
+
 # 计划
 # 创建一个全局定时器用来检测种子下载情况
 # 群文件上传成功回调
@@ -160,7 +163,6 @@ async def check_down_status(
     qb = await get_qb_client()
     if not qb:
         return
-    scheduler = require("nonebot_plugin_apscheduler").scheduler
     # 防止中途删掉任务，无限执行
     try:
         info = qb.get_torrent(hash_str)
@@ -223,7 +225,6 @@ async def delete_msg(bot: Bot, msg_ids: List[Dict[str, Any]]) -> None:
 
 
 async def rss_trigger(bot: Bot, hash_str: str, group_ids: List[str], name: str) -> None:
-    scheduler = require("nonebot_plugin_apscheduler").scheduler
     # 制作一个频率为“ n 秒 / 次”的触发器
     trigger = IntervalTrigger(seconds=int(config.down_status_msg_date), jitter=10)
     job_defaults = {"max_instances": 1}

--- a/src/plugins/ELF_RSS2/rss_class.py
+++ b/src/plugins/ELF_RSS2/rss_class.py
@@ -2,7 +2,6 @@ import re
 from copy import deepcopy
 from typing import Any, Dict, List, Optional
 
-from nonebot.log import logger
 from tinydb import Query, TinyDB
 from tinydb.operations import set as tinydb_set
 from yarl import URL
@@ -193,26 +192,16 @@ class Rss:
         rss_old = Rss.read_rss()
         return [rss for rss in rss_old if user in rss.user_id]
 
-    def set_cookies(self, cookies_str: str) -> bool:
-        try:
-            cookies = {}
-            for line in cookies_str.split(";"):
-                if "=" in line:
-                    key, value = line.strip().split("=", 1)
-                    cookies[key] = value
-            self.cookies = cookies
-            with TinyDB(
-                JSON_PATH,
-                encoding="utf-8",
-                sort_keys=True,
-                indent=4,
-                ensure_ascii=False,
-            ) as db:
-                db.update(tinydb_set("cookies", cookies), Query().name == self.name)  # type: ignore
-            return True
-        except Exception:
-            logger.exception(f"{self.name} 的 Cookies 设置时出错！")
-            return False
+    def set_cookies(self, cookies: str) -> None:
+        self.cookies = cookies
+        with TinyDB(
+            JSON_PATH,
+            encoding="utf-8",
+            sort_keys=True,
+            indent=4,
+            ensure_ascii=False,
+        ) as db:
+            db.update(tinydb_set("cookies", cookies), Query().name == self.name)  # type: ignore
 
     def upsert(self, old_name: Optional[str] = None) -> None:
         with TinyDB(

